### PR TITLE
Cleanup code from old JLS versions

### PR DESCRIPTION
**Pull request by iloveeclipse**
Follow up on https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2761 and https://github.com/eclipse-jdt/eclipse.jdt.core/issues/3304

### CODE REVIEW

**Comment by stephan-herrmann** [comment]
I honestly don't know if it is a good idea to stir up more dust like this before we're done with the previous step, viz. #2761 and #2758. 

From scanning the changes it looks like every single change would need a review, since this is not a mechanical change. This would take a lot of time, when done thoroughly. That seems to contradict the goal to make our lives easier.

I would be much more relaxed if we did such removal only in locations that are being changed for some other reason. Then we know that the person making the change understands the code at hand and its peculiarities (so we hope), and reviewing is feasible. And legacy that is not being looked at doesn't cause any pain, does it?

**Comment by iloveeclipse** [comment]
> I honestly don't know if it is a good idea to stir up more dust like this before we're done with the previous step, viz. #2761 and #2758.

Note, the mentioned tickets are staying open because either they are about some corner cases where I need help from people with deep JDT knowledge or are about ancient tests written at Java 1.3 - 1.5 times which do not run with Java 1.8 and no one has an idea why. 

This is not something global that would prevent us from doing any further steps in cleaning up code base from pre-Java 8 support.

> From scanning the changes it looks like every single change would need a review, since this is not a mechanical change. This would take a lot of time, when done thoroughly. That seems to contradict the goal to make our lives easier.

I would do the review, the changes are mostly of a purely mechanical nature and require mostly knowledge of boolean algebra rules in a generic context of JDT. I believe I'm qualified for that job and I also have time right now.

> I would be much more relaxed if we did such removal only in locations that are being changed for some other reason.

Sure, but that would never end, if you look at the numbers in this commit (~10.000 lines of code removed an ~5.000 touched).

Also, if I would like to have relaxed life I wouldn't select JDT maintenance as a part of it :-). But that's life, sometimes one has to do things that are not pleasant.

> Then we know that the person making the change understands the code at hand and its peculiarities (so we hope), and reviewing is feasible. 

I'm surely don't understand the code in question, that is true. But I have enough experience for cleanup the code of such constructs like `if(false)` or `if(true)`, which is mostly the work I've did so far and I believe it is going well forward, way faster and easier as original change in https://github.com/eclipse-jdt/eclipse.jdt.core/pull/2551.

> And legacy that is not being looked at doesn't cause any pain, does it?

It depends. The final trigger for me was a bug report where I've looked into the code and found myself in a complex spaghetti code enriched with the (dead) blocks of code guarded by "if Java < 1.4 or Java < 1.7" etc. Honestly, the code is complex enough without these if/else constructs and blocks of dead code following them.

**Comment by iloveeclipse** [comment]
The build is green. I will now squash all changes except one which I believe deserves extra commit (
[Use ClassFileConstants.getLatestJDKLevel() in SearchPattern by default](https://github.com/eclipse-jdt/eclipse.jdt.core/pull/3883/commits/80992bf98c60cf553891f21a7cccc22fc487b378)).

**Comment by iloveeclipse** [comment]
Correction, last batch of test changes was found locally :-) https://github.com/eclipse-jdt/eclipse.jdt.core/pull/3883/commits/fbe4099f76c74692e7a0bc8c417ad32c88fefb03
But now to squashing...

**Comment by iloveeclipse** [comment]
@stephan-herrmann , @srikanth-sankaran , @jarthana , @mpalat : I plan to review my code changes next week. I don’t expect anyone of you to perform a full review of changes here, it is too much. The nature of changes here is really very simple, in most cases simply removing dead conditions and following dead code branches. I did it all manually, and it was pretty easy exercise, as you remember, original work for 1.8 minimal level took weeks with lot of non trivial debugging. Now I needed two days and we have a green state.

If you have time and are interested, you can review some parts of the code that in your opinion deserves a closer look. Anyway, I plan to review production code changes during next week and plan to merge this change next Friday, of course if there would be no issues found during review. Have a nice weekend.

**Comment by srikanth-sankaran** [comment]
> @stephan-herrmann , @srikanth-sankaran , @jarthana , @mpalat : I plan to review my code changes next week. I don’t expect anyone of you to perform a full review of changes here, it is too much. The nature of changes here is really very simple, in most cases simply removing dead conditions and following dead code branches. I did it all manually, and it was pretty easy exercise, as you remember, original work for 1.8 minimal level took weeks with lot of non trivial debugging. Now I needed two days and we have a green state.
> 
> If you have time and are interested, you can review some parts of the code that in your opinion deserves a closer look. Anyway, I plan to review production code changes during next week and plan to merge this change next Friday, of course if there would be no issues found during review. Have a nice weekend.

I went back and looked at the discussion in https://github.com/eclipse-jdt/eclipse.jdt.core/issues/3304 

I will say that I am torn, seeing value in and agreeing with points raised on both sides.

In https://github.com/eclipse-jdt/eclipse.jdt.core/issues/3304#issuecomment-2474892505 I offered to take the ownership for cleaning up the remaining disabled tests but unfortunately haven't made much (any) progress - sorry about that!, other than trying to palm it off to @jespersm :) who made some progress (thanks!)

Is there a via-media solution here ? Given this is not a user visible feature that is blocking anyone, if this can be deferred to 4.36 M3, I can prioritize both (a) the cleanup/deletion/enabling of disabled tests and (b) getting this thoroughly reviewed and integrated and also (c) take responsibility for keeping the PR up-to-date in the mean time, so the painstaking work is not wasted.

I have a good shot at wrapping up the major tasks for Records 2.0 in time for M2 (https://github.com/eclipse-jdt/eclipse.jdt.core/issues/1090, https://github.com/eclipse-jdt/eclipse.jdt.core/issues/3893, https://github.com/eclipse-jdt/eclipse.jdt.core/issues/3901 and friends) and that would free me up for M3 work on these tasks.

Thoughts ?

**Comment by srikanth-sankaran** [comment]
> I would be much more relaxed if we did such removal only in locations that are being changed for some other reason. 

This would unfortunately clutter up many PRs. I prefer one sweeping change across the code base but with due diligence like code reviews and stress testing - IBM folk may be requested to test on their internal software for example. 

I also detest these blocks lingering longer than they need to - maybe it is all in the mind - I get frustrated by the needless clutter 🙂

**Comment by iloveeclipse** [comment]
> Is there a via-media solution here ? 

The solution for **real** problem is this PR.

Which solutions for which problems do we want more?

What did we want? Wait for [~19 disabled tests](https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2758) (out of 114582 passed) to be analyzed? Is there **any** proof that these 19 tests test code not tested by succeeded 114582 tests and if fixed, would uncover compiler issues? If so, let assume we have 19 real bugs to be fixed. Why are these bugs more important to solve compared to **[74 known](https://github.com/eclipse-jdt/eclipse.jdt.core/issues?q=is%3Aissue%20state%3Aopen%20label%3Abug)** bugs reported for JDT? And many more bugs **automatically** closed in bugzilla because no one had time to investigate them? Why should 19 **potential** issues prevent this PR from merge but 79 **known** issues don't? Are they more important or urgent?

> Given this is not a user visible feature that is blocking anyone

It is blocking **me**: if we postpone this to any later time, I would have to rebase & deal with merge conflicts, wasting my time on that instead of working on other issues. Also I can't really see this dead code anymore, I really want a cleaner codebase.

> if this can be deferred to 4.36 M3, I can prioritize both (a) the cleanup/deletion/enabling of disabled tests and (b) getting this thoroughly reviewed and integrated and also (c) take responsibility for keeping the PR up-to-date in the mean time, so the painstaking work is not wasted.

Regarding (b) and (c): The one reason why **I personally** wanted to do this work is to offload everyone with more JDT knowledge from doing this time consuming and boring work, so you could spend your time on solving **real** problems. So I don't want **anyone** from the people on CC to spend **any** extra time dealing with Java < 1.8 **anymore**, because we have enough problems with Java >= 1.8 and that is where the project resources should be spent.

Regarding (a): we should have more people interesting in fixing issues, so where they are? Obviously priorities are different...

Once again: what are the objections raised?

> I honestly don't know if it is a good idea to stir up more dust like this before we're done with the previous step, viz. 
> https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2761 and https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2758.

1) From https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2761: we have 19 disabled tests, two **existing** issues found during transition to 1.8, some opened questions no one has answers so far. I don't see how these are related to the proposed change here, that **only removes dead code**.
2) General objection against mass PR's because it is impossible to review. This is fully understandable and I personally dislike mass changes too! But I didn't hear any proposals how we can change ~23.000 lines of code (see this PR data above) in any "reasonable" way. Assuming we would split this PR to "smaller" chunks of 100 lines per PR, we would need 230 PR's, which means, **one PR every work day for the whole year**. Do we have two engineers (one creating the PR and another one reviewing it) free for this work? Can anyone commit to do this? The clear answer is **NO!**
3) General fear that mass changes would introduce regressions. Absolutely reasonable point. During my work for Eclipse platform I spend half of my time hunting & fixing regressions and of course I know best about that possibility. But we have luck in this case: 
- The changes in question are really trivial (removing dead boolean conditions and following dead code)
- mostly about black/white decisions in the code, means if the wrong condition is chosen or wrong code branch removed, we **should** see a regression.
- done by an experienced engineer
- backed by 114582 **successful** tests
- there is company (Advantest) that **commits** the time of two JDT committers with enough experience in JDT to fix all the regressions that will be found after merging of this PR.

With that I still believe that: 

- it is good idea to remove dead code in one big bang
- the best time to do it is now before M2, given that the change is there, all tests are green and no major code rework is started on Java 25 support branch yet
- if there will be no conclusion till Friday, I would act as a project lead and merge this PR except there would be raised objections that are not on the list yet or were not discussed yet in the tickets before.

**Comment by jarthana** [comment]
I am for the general idea of removal. However, I also share Stephan's concerns of reviewing all the changes can be very difficult given how busy JDT people in general are. Perhaps, we can share this review effort based on our expertise? I volunteer to review all things related to model and APT. Or perhaps can this PR even be split based on component?

> What did we want? Wait for [~19 disabled tests](https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2758) (out of 114582 passed) to be analyzed? Is there **any** proof that these 19 tests test code not tested by succeeded 114582 tests and if fixed, would uncover compiler issues? If so, let assume we have 19 real bugs to be fixed. Why are these bugs more important to solve compared to **[74 known](https://github.com/eclipse-jdt/eclipse.jdt.core/issues?q=is%3Aissue%20state%3Aopen%20label%3Abug)** bugs reported for JDT? And many more bugs **automatically** closed in bugzilla because no one had time to investigate them? Why should 19 **potential** issues prevent this PR from merge but 79 **known** issues don't? Are they more important or urgent?

FWIW, I had a quick look at the disabled tests and looks like most are related to non-compiler components, which we all know haven't had proper attention or a full time owner for a while now. Having said that, commented out tests *probably* means that they shouldn't/wouldn't have been commented out in the first place if the committer had enough time.

**Comment by iloveeclipse** [comment]
> I also share Stephan's concerns of reviewing all the changes can be very difficult given how busy JDT people in general are.

To be honest, **most** of the merged JDT PR's are **not reviewed**, although they are not about trivial code changes like this one. 

- All merged PR's: 1505: https://github.com/eclipse-jdt/eclipse.jdt.core/pulls?q=is%3Apr+is%3Amerged+
- PR's merged **without review**: **1230** : https://github.com/eclipse-jdt/eclipse.jdt.core/pulls?q=is%3Apr+is%3Amerged+review%3Anone+
- PR is merged **with** approved review: **14**: https://github.com/eclipse-jdt/eclipse.jdt.core/pulls?q=is%3Apr+is%3Amerged+review%3Aapproved

So why do we apply double standards on this particular PR?

**Comment by jarthana** [comment]
> So why do we apply double standards on this particular PR?

That's not my intention at all. Just that, the size of the patch, the breadth it covers and what a seemingly trivial change can cause make me think that it won't hurt to have the respective stakeholders take a look at the changes.

**Comment by iloveeclipse** [comment]
> it won't hurt to have the respective stakeholders take a look at the changes.

That was [my intention too](https://github.com/eclipse-jdt/eclipse.jdt.core/pull/3883#issuecomment-2779059807), sure. 
I just don't think you really want to review **all** of the changes, given the changes nature & number.

**Comment by srikanth-sankaran** [comment]
> > I also share Stephan's concerns of reviewing all the changes can be very difficult given how busy JDT people in general are.
> 
> To be honest, **most** of the merged JDT PR's are **not reviewed**, although they are not about trivial code changes like this one.
> 
> * All merged PR's: 1505: https://github.com/eclipse-jdt/eclipse.jdt.core/pulls?q=is%3Apr+is%3Amerged+
> * PR's merged **without review**: **1230** : https://github.com/eclipse-jdt/eclipse.jdt.core/pulls?q=is%3Apr+is%3Amerged+review%3Anone+
> * PR is merged **with** approved review: **14**: https://github.com/eclipse-jdt/eclipse.jdt.core/pulls?q=is%3Apr+is%3Amerged+review%3Aapproved
> 
> So why do we apply double standards on this particular PR?

OK, You points are valid and I will withdraw my proposal to defer this. I am OK with our going forward with this, subjecting to a go from @jarthana after some stress testing. 

I might add a genuine compliment here that you have an alternate career as a lawyer where you will be very successful if you ever choose to pursue that 🤣 😆

**Comment by srikanth-sankaran** [comment]
> And many more bugs **automatically** closed in bugzilla because no one had time to investigate them?

That was the weirdest thing in my experience! 

Elsewhere I proposed a mechanism to capture current (even if faulty) behavior of the compiler on every incoming defect into a junit of a name like CurrentButPotentiallyWrongBehaviorTests.java - that way when we propose a PR to fix a bug, we will know all other defects where there is a change in behavior due to the current PR - that would allow us to close all stale defects which got fixed by the current PR. But this hasn't taken off due basically to no good reason - my bad.

**Comment by srikanth-sankaran** [comment]
> Elsewhere I proposed a mechanism to capture current (even if faulty) behavior of the compiler on every incoming defect into a junit

This was proposed here: https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2306#issuecomment-2046502875

@stephan-herrmann has actually followed up here with https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2306#issuecomment-2053595157

So all of us would do well to follow suit and add to this suite whenever we triage ...

**Comment by srikanth-sankaran** [comment]
> > Elsewhere I proposed a mechanism to capture current (even if faulty) behavior of the compiler on every incoming defect into a junit
> 
> This was proposed here: [#2306 (comment)](https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2306#issuecomment-2046502875)
> 
> @stephan-herrmann has actually followed up here with [#2306 (comment)](https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2306#issuecomment-2053595157)
> 
> So all of us would do well to follow suit and add to this suite whenever we triage ...

To be clear, I am saying this could significantly reduce the build up of stale defects causing a github-tracker-genie from being invented to automatically close issues in some distant future - god forbid!

**Comment by stephan-herrmann** [comment]
Did I overreact (again) by raising concerns here? Perhaps.

Two aspects that triggered me:
* the unfinished #2761 feels like a regression, and "normally" I expect everybody causing a regression to fix it before starting any other work.
* we have been hit before by changes that looked trivial but did introduce new bugs.

Of course @iloveeclipse is not the hit-and-run kind of committer, letting others clean-up after them. So I might as well trust your judgement of what benefits JDT.

Regarding incremental vs big-bang I will only add, that I don't see clean-up parts "cluttering" other PRs, **if** we simply keep clean-up bits in a separate commit (within the PR that motivates it). OTOH, seeing that even among compiler devs there is a tendency towards big-bang removal, so be it.

**Comment by stephan-herrmann** [comment]
> What did we want? Wait for [~19 disabled tests](https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2758) (out of 114582 passed) to be analyzed? Is there **any** proof that these 19 tests test code not tested by succeeded 114582 tests and if fixed, would uncover compiler issues? If so, let assume we have 19 real bugs to be fixed. Why are these bugs more important to solve compared to **[74 known](https://github.com/eclipse-jdt/eclipse.jdt.core/issues?q=is%3Aissue%20state%3Aopen%20label%3Abug)** bugs reported for JDT? And many more bugs **automatically** closed in bugzilla because no one had time to investigate them? Why should 19 **potential** issues prevent this PR from merge but 79 **known** issues don't? Are they more important or urgent?

Good questions.

We don't have a formal system of prioritizing open bugs. 

For transparency, here are a few criteria that I use:

High priority
* regressions should be fixed while memory of the commit causing it is still fresh
* new Java features should be implemented before the corresponding Java release
* bugs that may cause loss of data (e.g., due to OOME or deadlock)

Not so high priority
* bugs without a reproducer
* bugs involving raw types (legacy with known differences between JLS and javac)
* bugs where type inference involves wildcards / captures (known differences between JLS and javac, JLS may still be insufficient)

File: org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/problem/ProblemReporter.java [hunk 1]
```diff
@@ -3093,7 +3092,7 @@ public void illegalModifierForInterfaceField(FieldDeclaration fieldDecl) {
 }
 public void illegalModifierForInterfaceMethod(AbstractMethodDeclaration methodDecl, long  level) {
 
-	int problem = level < ClassFileConstants.JDK1_8 ? IProblem.IllegalModifierForInterfaceMethod :
```
Comment (mpalat) [inline org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/problem/ProblemReporter.java:3096]: IProblem.IllegalModifierForInterfaceMethod will be no longer used.  can be flagged for removal.
There could be other fields too.

**Review by mpalat — COMMENTED** [review]
+1
Did a "sampling" of around 50+ files. Also, tried the disassembler with an older classfile just to make sure that it works - no issues there as well/

**Comment by jarthana** [comment]
>  I am OK with our going forward with this, subjecting to a go from @jarthana after some stress testing.

I didn't find any issues with my testing. I also reviewed the files under org.eclipse.jdt.core and found things to be fine mostly. Just noted down few points when I went through the other changes:

1. `LookupTest.test013()` is failing at compliance 14 and 15. I guess Jenkins is passing because we are not testing at those levels.
2. Some tests still appear to be referencing older versions, for e.g., `JDK1_1` and `JDK1_2` (`JavaProjectTests.testJdkLevelRoot()`, `AttachSourceTests.testRootPath11()` and a bunch of stuff in `EclipseJustification`)
3. There's this `SwitchTest.JDKLevelSupportingStringSwitch` field still referencing `JDK1_7`

**Comment by iloveeclipse** [comment]
> * Some tests still appear to be referencing older versions, for e.g., `JDK1_1` and `JDK1_2` (`JavaProjectTests.testJdkLevelRoot()`, `AttachSourceTests.testRootPath11()` and a bunch of stuff in `EclipseJustification`)

Yep, I've let few places where the tests expected **errors** by using old JLS target, and I avoided to touch `EclipseJustification` code because it wasn't clear to me at which extent it deals also to documentation of known JCj/javac discrepancies.

> * There's this `SwitchTest.JDKLevelSupportingStringSwitch` field still referencing `JDK1_7`

Same here, on the first glance that looked for me having some documentation value but now I went down to the test code and see it was using as a simple guard. I will fix that.

File: org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/problem/ProblemReporter.java (hunk 1 above)
Comment (iloveeclipse) [reply org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/problem/ProblemReporter.java:3096]: > IProblem.IllegalModifierForInterfaceMethod will be no longer used. can be flagged for removal. 

Will be in the next patch.

> There could be other fields too.

Any idea how to detect them?

**Comment by iloveeclipse** [comment]
> 1. `LookupTest.test013()` is failing at compliance 14 and 15. I guess Jenkins is passing because we are not testing at those levels.

Also fails on master, not related to this change.

File: org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/problem/ProblemReporter.java (hunk 1 above)
Comment (iloveeclipse) [reply org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/problem/ProblemReporter.java:3096]: I've found some documented constants and removed them together with related ProblemReporter code via https://github.com/eclipse-jdt/eclipse.jdt.core/pull/3883/commits/b4abe6ed49bc1c19a760f6040f44c01142548a2f

**Comment by iloveeclipse** [comment]
@mpalat: could you please check https://github.com/eclipse-jdt/eclipse.jdt.core/pull/3883/commits/af83290c268e95d01e3c8e6f571c4e909b9efbac ?
This removes lot of problem related code that shouldn't be triggered anymore with 1.8+.

File: org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/problem/ProblemReporter.java (hunk 1 above)
Comment (iloveeclipse) [reply org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/problem/ProblemReporter.java:3096]: Found even more, see https://github.com/eclipse-jdt/eclipse.jdt.core/pull/3883/commits/af83290c268e95d01e3c8e6f571c4e909b9efbac

Comment (mpalat) [reply org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/problem/ProblemReporter.java:3096]: > 
> Any idea how to detect them?

I remember @srikanth-sankaran had used a trick - maynot be exactly the same but something similar : 

- Make all the IProblem Constants Private 
- The compiler will throw the error message about visibility being reduced on the constants that are used.
- Remove these  after a quick check whether it's related to 1.8 or not (even otherwise it would be a superset)

Assumption here is that this check is on the PR which has already removed unnecessary code.
well, my 2 cents... if it helps.

Comment (iloveeclipse) [reply org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/problem/ProblemReporter.java:3096]: Hmm, this didn't work at all. I hat to change interface to abstract class too, and only got errors but not about reduced visibility. Anyway, with last commit I've removed ~30 problems, that should be OK for now.

**Comment by srikanth-sankaran** [comment]
> @mpalat: could you please check [af83290](https://github.com/eclipse-jdt/eclipse.jdt.core/commit/af83290c268e95d01e3c8e6f571c4e909b9efbac) ? This removes lot of problem related code that shouldn't be triggered anymore with 1.8+.

It is customary to also zap out the diagnostic text from /org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/problem/messages.properties - these are dead data strings once the IProblem is not emitted anymore.

See the lines beginning with `###[obsolete]` in that file for example:

###[obsolete] 1600 = Incompatible switch results expressions {0}
###[obsolete] 1601 = A switch expression should have a non-empty switch block
###[obsolete] 1602 = A switch expression should have at least one result expression
###[obsolete] 1603 = A switch labeled block in a switch expression should not complete normally

**Comment by mpalat** [comment]
> @mpalat: could you please check [af83290](https://github.com/eclipse-jdt/eclipse.jdt.core/commit/af83290c268e95d01e3c8e6f571c4e909b9efbac) ? This removes lot of problem related code that shouldn't be triggered anymore with 1.8+.

@akurtakov - I've looked at the  commit. Could not find anything wrong. Looks good. Thanks!

File: org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/problem/ProblemReporter.java (hunk 1 above)
Comment (mpalat) [reply org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/problem/ProblemReporter.java:3096]: > Hmm, this didn't work at all. I hat to change interface to abstract class too, and only got errors but not about reduced visibility. Anyway, with last commit I've removed ~30 problems, that should be OK for now.

ok. guess I should have been more explicit here - The error is more from the API compatibility perspective

for example by flagging the following as `private`

`	private int IllegalModifierForClass = TypeRelated + 302;
`
you can see that I get an error and the following quick fix:
"create a compatibility filter:

Creates a compatibility problem filter with comment for API 
tools problem:

The visibility of the field 
org.eclipse.jdt.core.compiler.IProblem.IllegalModifierForClass 
has been reduced
"

image

**Comment by akurtakov** [comment]
> > @mpalat: could you please check [af83290](https://github.com/eclipse-jdt/eclipse.jdt.core/commit/af83290c268e95d01e3c8e6f571c4e909b9efbac) ? This removes lot of problem related code that shouldn't be triggered anymore with 1.8+.
> 
> @akurtakov - I've looked at the commit. Could not find anything wrong. Looks good. Thanks!

This is for @iloveeclipse I assume.

**Comment by iloveeclipse** [comment]
> It is customary to also zap out the diagnostic text from /org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/problem/messages.properties - these are dead data strings once the IProblem is not emitted anymore.

Thanks! 
I've updated the last commit now, see https://github.com/eclipse-jdt/eclipse.jdt.core/pull/3883/commits/804391fe0dd8040022391458878bf7d9bcf186ce

**Comment by mpalat** [comment]
> > > @mpalat: could you please check [af83290](https://github.com/eclipse-jdt/eclipse.jdt.core/commit/af83290c268e95d01e3c8e6f571c4e909b9efbac) ? This removes lot of problem related code that shouldn't be triggered anymore with 1.8+.
> This is for @iloveeclipse I assume.

oops. yes.. for @iloveeclipse

**Comment by iloveeclipse** [comment]
OK, I'm going to merge this. 

I will also trigger SDK build to see possible test regressions in JDT UI / Debugger and PDE tests. I've executed tests locally, but it is of course different to the regular SDK tests, so there still could be some test not executed or executed differently in the different build environment.

**Comment by iloveeclipse** [comment]
SDK tests look good in general on Java 21 Linux/Windows: https://download.eclipse.org/eclipse/downloads/drops4/I20250411-0200/testResults.php

The jdt model test on Linux Java 24 build are [known to be unstable](https://github.com/eclipse-jdt/eclipse.jdt.core/issues/3325).
